examples/numba_support.py

```diff
@@ -0,0 +1,973 @@
+        ("handle", "OptixTraversableHandle"),
+    )
+
+
+# "Declare" a global called params
+params = ParamsStruct()
+
+
+class Params(types.Type):
+    def __init__(self):
+        super().__init__(name="ParamsType")
+
+
+params_type = Params()
+
+
+# ParamsStruct data model
+
+
+@register_model(Params)
+class ParamsModel(models.StructModel):
+    def __init__(self, dmm, fe_type):
+        members = [
+            ("image", types.CPointer(uchar4)),
+            ("image_width", types.uint32),
+            ("image_height", types.uint32),
+            ("cam_eye", float3),
+            ("cam_u", float3),
+            ("cam_v", float3),
+            ("cam_w", float3),
+            ("handle", OptixTraversableHandle),
+        ]
+        super().__init__(dmm, fe_type, members)
+
+
+make_attribute_wrapper(Params, "image", "image")
+make_attribute_wrapper(Params, "image_width", "image_width")
+make_attribute_wrapper(Params, "image_height", "image_height")
+make_attribute_wrapper(Params, "cam_eye", "cam_eye")
+make_attribute_wrapper(Params, "cam_u", "cam_u")
+make_attribute_wrapper(Params, "cam_v", "cam_v")
+make_attribute_wrapper(Params, "cam_w", "cam_w")
+make_attribute_wrapper(Params, "handle", "handle")
+
+
+@typeof_impl.register(ParamsStruct)
+def typeof_params(val, c):
+    return params_type
+
+
+# ParamsStruct lowering
+# The below makes 'param' a global variable, accessible from any user defined
+# kernels.
+
+
+@lower_constant(Params)
+def constant_params(context, builder, ty, pyval):
+    try:
+        gvar = builder.module.get_global("params")
+    except KeyError:
+        llty = context.get_value_type(ty)
+        gvar = cgutils.add_global_variable(
+            builder.module, llty, "params", addrspace=nvvm.ADDRSPACE_CONSTANT
+        )
+        gvar.linkage = "external"
+        gvar.global_constant = True
+
+    return builder.load(gvar)
+
+
+# MissData
+# ------------
+
+# Structures as declared in triangle.h
+class MissDataStruct:
+    fields = ("bg_color", "float3")
+
+
+MissData = MissDataStruct()
+
+
+class MissData(types.Type):
+    def __init__(self):
+        super().__init__(name="MissDataType")
+
+
+miss_data_type = MissData()
+
+
+@register_model(MissData)
+class MissDataModel(models.StructModel):
+    def __init__(self, dmm, fe_type):
+        members = [
+            ("bg_color", float3),
+        ]
+        super().__init__(dmm, fe_type, members)
+
+
+make_attribute_wrapper(MissData, "bg_color", "bg_color")
+
+
+@typeof_impl.register(MissDataStruct)
+def typeof_miss_data(val, c):
+    return miss_data_type
+
+
+# MissData Constructor
+@type_callable(MissDataStruct)
+def type_miss_data_struct(context):
+    def typer(sbt_data_pointer):
+        if isinstance(sbt_data_pointer, SbtDataPointer):
+            return miss_data_type
+
+    return typer
+
+
+@lower(MissDataStruct, sbt_data_pointer)
+def lower_miss_data_ctor(context, builder, sig, args):
+    # Anyway to err if this ctor is not called inside __miss__* program?
```

Comment:
I learnt a little bit more that this cast in cpp (from `SBTDataPointer` to `MissData`) is possible because there is no `RayGenData` and `HitGroupData`. So this function here is also only specific to the triangle example. Ideally we should have an SBTRecord type based on the layout defined in `create_sbt` and generates the proper record type based on user's request. (e.g. sbtrecord.get_miss_data() or sbtrecord.get_ray_gen_data() etc.)